pkg/rtc/transport.go

```diff
@@ -327,20 +476,146 @@ func (t *PCTransport) AddICECandidate(candidate webrtc.ICECandidateInit) error {
 		return nil
 	}
 
-	t.params.Logger.Infow("add candidate ", "candidate", candidate.Candidate)
+	t.lock.RLock()
+	if t.preferTCP && !strings.Contains(candidate.Candidate, "tcp") {
+		t.lock.RUnlock()
+		t.params.Logger.Infow("filtering out remote candidate", "candidate", candidate.Candidate)
+		return nil
+	}
+	t.lock.RUnlock()
 
+	t.params.Logger.Infow("add candidate ", "candidate", candidate.Candidate)
```

Comment:
unrelated, but to make logs more readable, it'd be great to print all ICE candidates in the same log entry as an array. would be great to have both local and remote candidates in the same entry too.

Reply:
Okay, will see how to do that. Will look at it when I do caching candidates.